File: GO/2.go
package main

import (
	"bufio"
	"fmt"
	"math/rand"
	"os"
	"strconv"
	"time"
)

type pair struct {
	first  int
	second []string
}

type visitor struct {
	first  string
	second int
}

func sortByTime(visitors []visitor) {
	n := len(visitors)
	for i := 0; i < n-1; i++ {
		swapped := false
		for j := 0; j < n-1-i; j++ {
			if visitors[j].second < visitors[j+1].second {
				visitors[j], visitors[j+1] = visitors[j+1], visitors[j]
				swapped = true
			}
		}
		if !swapped {
			break
		}
	}
}

func sim() {
	fmt.Println("Команды")
	fmt.Println("ENQUEUE <время>")
	fmt.Println("DISTRIBUTE")

	rand.Seed(time.Now().UnixNano())

	fmt.Print("Введите кол-во окон\n ")
	var windowsCount int
	fmt.Scanln(&windowsCount)

	windows := make([]pair, windowsCount)
	for i := range windows {
		windows[i].first = 0
		windows[i].second = []string{}
	}

	var visitors []visitor
	var ticket_counter = 1

	scanner := bufio.NewScanner(os.Stdin)
	scanner.Split(bufio.ScanWords)

READ_LOOP:
	for {
		if !scanner.Scan() {
			break
		}
		cmd := scanner.Text()

		switch cmd {
		case "ENQUEUE":
			if !scanner.Scan() {
				break
			}
			durationStr := scanner.Text()
			duration, err := strconv.Atoi(durationStr)
			if err != nil || duration < 0 {
				continue
			}
			ticket_num := rand.Intn(1000)
			ticket := "T" + strconv.Itoa(ticket_num)
			visitors = append(visitors, visitor{ticket, duration})
			fmt.Println(ticket)
			ticket_counter++

		case "DISTRIBUTE":
			break READ_LOOP

		default:
			fmt.Printf("Неизвестная команда: %s\n", cmd)
		}
	}

	sortByTime(visitors)

	for _, visitor := range visitors {
		minTime := int(^uint(0) >> 1) // INT_MAX emulation
		bestWindow := 0
		for i, window := range windows {
			if window.first < minTime {
				minTime = window.first
				bestWindow = i
			}
		}
		windows[bestWindow].first += visitor.second
		windows[bestWindow].second = append(windows[bestWindow].second, visitor.first)
	}

	for i := 0; i < windowsCount; i++ {
		fmt.Printf("Окно %d (%d минут): ", i+1, windows[i].first)
		if len(windows[i].second) > 0 {
			for j, ticket := range windows[i].second {
				if j > 0 {
					fmt.Print(", ")
				}
				fmt.Print(ticket)
			}
		}
		fmt.Println()
	}
}

File: C++/2.cpp
#include <iostream>
#include <vector>
#include <queue>
#include <string>
#include <sstream>
#include <iomanip>
#include <cstdlib>
#include <ctime>
#include <climits>

using namespace std;

void sortByTime(vector<pair<string, int>>& visitors) {
    int n = visitors.size();
    for (int i = 0; i < n - 1; ++i) {
        bool swapped = false;
        for (int j = 0; j < n - 1 - i; ++j) {
            if (visitors[j].second < visitors[j + 1].second) {
                swap(visitors[j], visitors[j + 1]);
                swapped = true;
            }
        }
        if (!swapped) break;
    }
}

void sim() {
    cout << "Команды\nENQUEUE <время> \nDISTRIBUTE \n";

    srand(static_cast<unsigned int>(time(nullptr)));
    int windowsCount;
    cout << "Введите кол-во окон\n";
    cin >> windowsCount;

    vector<pair<int, vector<string>>> windows(windowsCount);
    for (auto& window : windows) {
        window.first = 0;
        window.second.clear();
    }

    vector<pair<string, int>> visitors;
    string cmd;
    int ticket_counter = 1;

    while (cin >> cmd) {
        if (cmd == "ENQUEUE") {
            int duration;
            cin >> duration;
            if (duration < 0) continue;

            int ticket_num = rand() % 1000;
            string ticket = "T" + to_string(ticket_num);

            visitors.emplace_back(ticket, duration);
            cout << ticket << "\n";
        }
        else if (cmd == "DISTRIBUTE") {
            break;
        }
    }

    sortByTime(visitors);
    for (auto visitor : visitors) {
        int minTime = INT_MAX;
        int bestWindow = 0;

        int index = 0;
        for (auto window : windows) {
            if (window.first < minTime){
                minTime = window.first;
                bestWindow = index;
            }
            index++;
        }

        windows[bestWindow].first += visitor.second;
        windows[bestWindow].second.push_back(visitor.first);
    }

    for (int i = 0; i < windowsCount; i++) {
        cout << "Окно " << (i + 1) << " (" << windows[i].first << " минут): ";
        if (!windows[i].second.empty()) {
            for (int j = 0; j < windows[i].second.size(); ++j) {
                if (j > 0) cout << ", ";
                cout << windows[i].second[j];
            }
        }
        cout << "\n";
    }
}
